### ISSUE TITLE
How should I adjust the hyperparameters to train a strong DDPM model based on CelebA for inpainting task?

### ISSUE THREAD

**Issue by LinWeiJeff** [jychoi118/P2-weighting#20]
I'm very grateful for your work about this github repository!!! However, I have some questions about the hyperparameters to want to ask you.
I know your repository is heavily based on [guided-diffusion](https://github.com/openai/guided-diffusion), and I want to use [guided-diffusion](https://github.com/openai/guided-diffusion) to train a DDPM model based on CelebA dataset consisting of 202,599 align&cropped human face images (each image is 218(height)x178(width) pixels) for human face image inpainting task.
Could you give me some suggestions about how to adjust the hyperparameters used in [guided-diffusion](https://github.com/openai/guided-diffusion) to get a strong DDPM model for human face image inpainting task?
Should I need to resize the images in the dataset I used to 256x256 firstly?

I know your repository adds two extra hyperparameters (one is `p2_gamma` and the another is `p2_k`), but I still want to use [guided-diffusion](https://github.com/openai/guided-diffusion) to train my own DDPM model firstly, so I come here to ask you how to adjust the other hyperparameters except for p2_gamma and p2_k.

I hope the trained strong DDPM model learns the features of human face well so it can be used for human face image synthesis task and human face image inpainting task (i.e. recovering the masked parts of a masked human face image).

I want to know how to adjust the values of the hyperparameters in **diffusion_defaults() of script_util.py**, **model_and_diffusion_defaults() of script_util.py** and **create_argparser() of image_train.py** in [guided-diffusion](https://github.com/openai/guided-diffusion) to train a strong denoising model in DDPM based on my CelebA dataset.

I have tried some combinations of the hyperparameters used in [guided-diffusion](https://github.com/openai/guided-diffusion) to train, however, the human face image inpainting results of the saved model files `ema_0.9999_XXX.pt` and `modelXXX.pt` are both bad.
I mainly used [RePaint](https://github.com/andreas128/RePaint) to perform sampling for human face image inpainting task, and as described in its README, the pretrained model [celeba256_250000.pt](https://drive.google.com/uc?id=1norNWWGYP3EZ_o05DmoW1ryKuKMmhlCX) (it is downloaded from [download.sh](https://github.com/andreas128/RePaint/blob/main/download.sh)) that [RePaint](https://github.com/andreas128/RePaint) used to perform sampling for human face image inpainting task is big (about 2.1GB) and its sampling result is not bad. However, I don't know why the model's size is so big and how it is trained.
In addition, because of the limitation of my GPU memory, I set the value of the hyperparameter `num_channels` only 64, I want to know if this hyperparameter affects the performance of the traind DDPM model. Should I try to set it larger?

In conclusion, I hope you can give me some suggestions about how to adjust the hyperparameters used in [guided-diffusion](https://github.com/openai/guided-diffusion) to get a strong DDPM model for human face image inpainting task.

Thanks a lot for your any help!!!
p.s. I directly and manually set up the values of the hyperparameters through the codes of [guided-diffusion](https://github.com/openai/guided-diffusion) not through flags.
